kernel/src/vm/vmar/mod.rs

```diff
@@ -1171,6 +1235,15 @@ impl<'a> VmarMapOptions<'a> {
         self
     }
 
+    /// Sets whether to allocate the mapping from high address to low address.
+    ///
+    /// The default value is true.
+    #[expect(dead_code)]
```

Comment:
Then `alloc_free_region` is useless. I think you should just remove `alloc_free_region`, rename `alloc_free_region_topdown` to `alloc_free_region`, and remove all `topdown`-related variables/methods.